src/api/v2/quickPickWizard/CreateQuickPickStep.ts

```diff
@@ -0,0 +1,40 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { IAzureQuickPickItem } from '@microsoft/vscode-azext-utils';
+import * as vscode from 'vscode';
+import { localize } from '../../../utils/localize';
+import { ResourceModelBase } from '../v2AzureResourcesApi';
+import { ContextValueFilter } from './ContextValueFilter';
+import { GenericQuickPickStep } from './GenericQuickPickStep';
+import { QuickPickWizardContext } from './QuickPickWizardContext';
+
+type CreateCallback = () => vscode.ProviderResult<never>;
+
+export class CreateQuickPickStep<TModel extends ResourceModelBase> extends GenericQuickPickStep<TModel> {
+    public constructor(treeDataProvider: vscode.TreeDataProvider<TModel>, contextValueFilter: ContextValueFilter, private readonly createCallback: CreateCallback) {

```

Comment:
Also might be worth having a contextExcluder.  The perfect example is for Logic Apps vs. Function Apps.  Both have the same kind, but Logic Apps has an extra kind, `workflowapp`.  However, we wouldn't want to have _exact_ matching because Function apps can have both `functionapp` and `functionapp,linux`

Reply:
We keep on running into this issue with type and kind all over the place. 

We should try to build on top of https://github.com/microsoft/vscode-azuretools/pull/1191 so that we don't have to solve the problem again here.